### ISSUE TITLE
Stop "supporting" more than two major revisions.

### ISSUE THREAD

**Issue by dsmiley** [docker-solr/docker-solr#337]
See the discussion here: https://github.com/docker-library/official-images/pull/8530#discussion_r469892206
> If we cut out anything older than 7.7.x, we should add a warning notice somewhere that these versions are stale, unsupported and likely not secure.

Yes, I think we should do this.  It's consistent with Apache Solr.  If say there's some big bad security bug in Solr for 6.whatever then Solr will get a 6.whatever update and we can do that for docker-solr as well, but that's unusual.  If the security bug is in the JDK... well... I'm not sure it's reasonable for projects to keep releasing updates to old versions to swap in new FROM images.  That seems more like a fundamental limitation in Docker itself -- not allowing you to run image XYZ while swapping out one of the layers (and all it's sub-layers) for another image (and it's sub-layers).  I wish you could!  People should be able to use JDK 15 as soon as it is out if they could.... but they can't (with Docker) without building their own images.  I digress :-)

**Comment by aadrian** [comment]
We are using Docker Solr exactly because it has older versions.

**Comment by dsmiley** [comment]
Quoting the discussion at docker-library:
> but some users still need the older versions/tags, so we don't actively delete them unless the image maintainer explicitly requests that we do so.

So 6.x and 5.x would still be available to you to continue to use; they just wouldn't be updated in normal circumstances.

**Comment by aadrian** [comment]
@dsmiley that's not the same thing like updating them with the latest JDKs and OS.

**Comment by janhoy** [comment]
> @dsmiley that's not the same thing like updating them with the latest JDKs and OS.

True, but if you are concerned with security you won’t be running Solr 5.x in the first place :-) You can check out this repo and build your custom 5.x image with whatever version of Linux and JRE you need.

Perhaps it will be a benefit that the older images will have older build dates on docket hub so it becomes obvious that you’re on your own?

**Comment by aadrian** [comment]
> True, but if you are concerned with security you won’t be running Solr 5.x in the first place :-) 

Many companies use in production older versions of Solr because newer ones are not 100% backwards compatible (drop in replacement) and would require development effort.

Another reason is that Solr itself is used by some other ready project (e.g. Nutch) that can't be just updated without side effects, so people are forced to keep using the older Solr versions.

**Comment by janhoy** [comment]
Perhaps we can continue to refresh our non-official images (https://hub.docker.com/r/dockersolr/docker-solr/tags) for every release? These are the same images, but with another tag, e.g. `dockersolr/docker-solr:8.6`, possibly in the future `apache/solr:8.6`, and are not built or QA'ed by Docker. Wdyt @dsmiley ? Perhaps we even stop publishing "official" images and teach people to `pull apache/solr` instead of `pull solr` in the future? This will be all up to the new Solr TLP to decide and integrate into release management procedures.

**Comment by dsmiley** [comment]
I don't have a strong opinion on any of this.  I'd _prefer_ to continue to publish "official" images because it's not clear what's gained by ceasing to do so.

**Comment by janhoy** [comment]
Good question. Sounds like Docker folks want us to conserve build resources by limiting to what is actively supported/used. Kind of makes sense. Perhaps include 6.6 as well and add documentation?

**Comment by tianon** [comment]
It is a burden on our build resources yes, but in the case of supported things, that's the reason for the program, so that's not really as large a concern as the optics for users -- by continuing to maintain/list these older releases, they're noted as "Supported" (and indeed, from an "image freshness" perspective, they appear to be!) which encourages users to use them thinking that they're still actively supported (not giving them the indication you noted above that if they care about security, they shouldn't actually be using Solr 5.x).

IMO https://lucene.apache.org/solr/downloads.html#about-versions-and-support also has a pretty compelling argument here:

> Apache Solr is under active development with frequent feature releases on the current major version. The previous major version will see occasional critical security- or bug fixes releases. Older versions are considered EOL (End Of Life) and **will not be further updated**. For this reason it may also be difficult to obtain community support for EOL versions.

(emphasis mine)

Removing these older versions would bring the image description (and the supported/updated reality) of https://hub.docker.com/_/solr in line with what's officially documented/supported at https://lucene.apache.org/solr/downloads.html.

**Comment by dsmiley** [comment]
I want to revisit this after @tianon recently prodded me on this: 
https://github.com/docker-library/official-images/pull/11485#issuecomment-991836483

In particular, I propose removing Solr 5 & Solr 6 and when 9 is released, Solr 7 as well.  They will _continue_ to be available on Docker hub but not mislabeled as "supported".  If some truly nasty bug affects those Solr versions, I assume we retain some recourse to do something about it.  Even the recent nasty Log4j problem thing is easily remedied by any user simply by adding a Java system property, which Solr makes easy to do.

**Comment by janhoy** [comment]
There shouldn't be any reason to keep refreshing 8.0, 8.1, 8.2, 8.3, 8.4, 8.5, 8.6, 8.7, 8.8, 8.9 and 8.10 either, as those individual minor releases are not patched. Imho, the only version listed as supported on hub should be `8.11.0` and perhaps `7.7.3`, but since 7.7.3 is 1,5 years old I'd rather not fool users into thinking that it is a fully patched version of 7.x.

I.e. perhaps our future image publishing policy should be to refresh only the tags of the actual release version. And only refresh latest previour-major version during all releases (to get linux and jdk updates), if we intend to keep that Solr version patched.

**Comment by tianon** [comment]
From https://github.com/docker-solr/docker-solr/pull/403 it sounds like the plan might be to wait until 9.0 to trim down the "supported" list?  Is that accurate or have I misread? :see_no_evil: :innocent:

**Comment by janhoy** [comment]
Yes, in a couple of weeks we'll have 9.0 out and then only 9.0 and 8.11 will be supported and refreshed. We are also re-organizing the whole release process and git repos, so you'll see some action soon :)
